File: day1/main.rb
input_file = './input.txt'

left_list = []
right_list = []

File.foreach(input_file) do |line|
  left, right = line.chomp.split
  left_list << left.to_i
  right_list << right.to_i
end

left_list.sort!
right_list.sort!

acc = 0
left_list.each_with_index do |left, i|
  acc += (left - right_list[i]).abs
end
# acc = left_list.zip(right_list).reduce(0) do |a, l|
  # a + (l[0] - l[1]).abs
# end 

print "Distance : #{acc}\n"

File: day1/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"math"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("./input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var left []int
	var right []int

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)

		l, _ := strconv.Atoi(fields[0])
		left = append(left, l)

		r, _ := strconv.Atoi(fields[1])
		right = append(right, r)
	}

	sort.Ints(left)
	sort.Ints(right)

	acc := 0.0
	for i, l := range left {
		acc += math.Abs(float64(l - right[i]))
	}

	fmt.Println("Disance : ", int(acc))
}
